src/getServerVersion.ts

```diff
@@ -37,7 +42,11 @@ async function validateCurrentVersion(
 
   const checkForUpdate = async () => {
     if (suggestUpgradeSetting?.value) {
-      return needCheckForUpdates(serverVersion, suggestUpgradeSetting.target);
+      return needCheckForUpdates(
+        serverVersion,
+        suggestUpgradeSetting.target,
```

Comment:
If I understand correctly, depending on where user set `checkForUpdates` switch then version will be stored in that scope (global vs workspace), I'm right? 
Here, because it's a simple boolean there are no such cases like with server version itself (where empty string in workspace is treated as a valid value and global setting is discarded)

Reply:
> If I understand correctly, depending on where user set checkForUpdates switch then version will be stored in that scope (global vs workspace), I'm right?

Yes, you're right!

> Here, because it's a simple boolean there are no such cases like with server version itself

What do you mean?

Do you mean, we should check for updating both Global and Workspace settings, even if `suggetUpgradeSetting` is `true` only in `Global` or `Workspace`?
In that case, I agree, but I think we can work on that in another issue / PR https://github.com/scalameta/metals-vscode/issues/1071

Reply:
@tanishiking I honestly don't remember what I was thinking and what I meant 😅

Reply:
haha, no worries! sorry for my delayed reply